src/DistributedLock.Tests/Infrastructure/Redis/RedisServer.cs

```diff
@@ -35,7 +35,12 @@ private RedisServer(int? port, bool allowAdmin)
                 .RedirectStandardErrorTo(Console.Error);
             ActiveServersByPort.Add(this.Port, this);
         }
-        this.Multiplexer = ConnectionMultiplexer.Connect($"localhost:{this.Port}{(allowAdmin ? ",allowAdmin=true" : string.Empty)}");
+        this.Multiplexer = ConnectionMultiplexer.Connect(new ConfigurationOptions
+        {
+            EndPoints = { { "localhost", this.Port } },
+            AllowAdmin = allowAdmin,
+            BacklogPolicy = BacklogPolicy.FailFast, // for backward compatibility with pre-2.5 versions
```

Comment:
Some tests started failing after upgrading to 2.5.27+ as noted [here](https://github.com/madelson/DistributedLock/issues/162#issuecomment-1637258292). I believe this is because of the following update (see the following excerpt from the [release notes](https://stackexchange.github.io/StackExchange.Redis/ReleaseNotes#2527-prerelease)):
> 2.5.27 (prerelease)
> - Adds: a backlog/retry mechanism for commands issued while a connection isn’t available ([#1912 by NickCraver](https://github.com/StackExchange/StackExchange.Redis/pull/1912))
>   - Commands will be queued if a multiplexer isn’t yet connected to a Redis server.
>   - Commands will be queued if a connection is lost and then sent to the server when the connection is restored.
>   - All commands queued will only remain in the backlog for the duration of the configured timeout.
>   - To revert to previous behavior, a new ConfigurationOptions.BacklogPolicy is available - old behavior is configured via options.BacklogPolicy = BacklogPolicy.FailFast. This backlogs nothing and fails commands immediately if no connection is available.

Reply:
Interesting. Which tests fail without this? Based on the description I would expect it to be tests that reproduce things like connection failure.

My biggest concern is whether the need for this setting means that the library won't work for users who _don't_ turn this on. Thoughts?

Finally, would be good to have a concise summary of the information in your comment + my questions above in the source itself.

Reply:
For (my) future reference, I just updated `StackExchange.Redis`  to 2.5.43 and noticed the following two failures:

> **Core_ReaderWriterAsMutex_RedisReaderWriter_Redis2x1Database_RedisSynchronizationStrategy_Redis2x1Database_RedisSynchronizationStrategy_Redis2x1DatabaseTest**
   TestParallelism
   Source: DistributedLockCoreTestCases.cs line 162
   Duration: 30 sec
  Message: 
  Timed out! (only 3/100 completed)
  Expected: not same as <System.Threading.Tasks.Task+DelayPromise>
  But was:  <System.Threading.Tasks.Task+DelayPromise>

> **Core_Redis_Redis2x1Database_RedisSynchronizationStrategy_Redis2x1DatabaseTest**
  TestParallelism
   Source: DistributedLockCoreTestCases.cs line 162
   Duration: 30 sec
  Message: 
  Timed out! (only 0/100 completed)
  Expected: not same as <System.Threading.Tasks.Task+DelayPromise>
  But was:  <System.Threading.Tasks.Task+DelayPromise>

Note that the number of completed tests can vary.

Reply:
I ended up figuring out what I think is a good path forward here: https://github.com/madelson/DistributedLock/blob/release-2.4/src/DistributedLock.Redis/RedLock/RedLockAcquire.cs#L155

Appreciate you bringing this to my attention.